platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/ProjectService.java

```diff
@@ -805,15 +805,37 @@ public ImportResponse importProject(@ApiParam(value = "Workspace ID", required =
         // Not all importers uses virtual file system API. In this case virtual file system API doesn't get events and isn't able to set
         // correct creation time. Need do it manually.
         long creationDate = System.currentTimeMillis();
+        
+        boolean folderExistedBeforeCommand = virtualEntryExists(workspace, path);
         VirtualFileEntry virtualFile = getVirtualFile(workspace, path, force);
+        FolderEntry baseProjectFolder = null;
 
-        final FolderEntry baseProjectFolder = (FolderEntry)virtualFile;
-        importer.importSources(baseProjectFolder, projectSource.getLocation(), projectSource.getParameters(), outputOutputConsumerFactory);
+        try {
+            baseProjectFolder = (FolderEntry)virtualFile;
+            importer.importSources(baseProjectFolder, projectSource.getLocation(), projectSource.getParameters(), outputOutputConsumerFactory);
+
+            //project source already imported going to configure project
+            return configureProject(importProject, baseProjectFolder, workspace, creationDate);
+        } catch (Exception e) {
+            // if import failed for some reason and the folder was created for this import
+            // remove it
+            if(!folderExistedBeforeCommand && baseProjectFolder != null) {
+                try {
+                   baseProjectFolder.remove();
+                } catch(Exception removeException) {
+                   LOG.error(String.format("An exception was thrown while trying to remove folder %s", baseProjectFolder.getPath()), removeException);
+                }
+            }
+            throw e;
+        }
 
-        //project source already imported going to configure project
-        return configureProject(importProject, baseProjectFolder, workspace, creationDate);
     }
 
+    private boolean virtualEntryExists(String workspace, String path) throws ForbiddenException, ServerException, NotFoundException {
+        VirtualFileEntry virtualFile = projectManager.getProjectsRoot(workspace).getChild(path);
+        return (virtualFile != null);
```

Comment:
we can use only one line
return projectManager.getProjectsRoot(workspace).getChild(path) != null